jupyterlitexblock/jupyterlitexblock.py

```diff
@@ -126,11 +141,33 @@ def studio_view(self, context=None):
         frag.initialize_js('JupterLiteXBlock')
         return frag
     
+    def delete_existing_files(self):
+        """
+        Delete existing files in the notebook folder.
+        """
+        folder_path = self.folder_base_path
+        if not self.storage.exists(folder_path):
+            return
+        
+        existing_files = self.storage.listdir(folder_path)[1]
+        for filename in existing_files:
+            file_path = f"{folder_path}/{filename}"
+            self.storage.delete(file_path)
+    
     def save_file(self, uploaded_file):
+        self.delete_existing_files()
         path = self.storage.save(f'{self.folder_base_path}/{uploaded_file.name}', ContentFile(uploaded_file.read()))
-        uploaded_file_url = self.storage.url(path)
+        url = self.storage.url(path)
+        if "https://" in url or "http://" in url:
+            uploaded_file_url = url
+        else:
+            scheme = "https" if settings.HTTPS == "on" else "http"
+            root_url = f'{scheme}://{settings.CMS_BASE}'
+            tmp_file = os.path.join(settings.MEDIA_ROOT, path)
```

Comment:
I don't think we need `MEDIA_ROOT` and `path` here. This should work
```suggestion
              uploaded_file_url = os.path.join(root_url, url)
```